modules/workbench/WORKBENCH-MODULE.md

```diff
@@ -0,0 +1,63 @@
+
+# Workbench Module
+The module directories here represent individually deployable subsystems, 
+microservices, or other functional units. It's easy enough to put all buckets, say,
+in a `gcs` module, but that wouldn't really let us operate on an individual components's bucket.
+
+Following is a broad outline fo each child module. If you feel irritated that you can't see, for example,
+all dashboards in one place, you can still go to the Console or use `gcloud`.
+
+A somewhat forward-looking plan for that would look like
+
+# Workbench Module Development Plan
```

Comment:
FWIW, here is my stack ranking of these:

Good improvements, if they can be achieved easily:
+ Sumologic
+ Reporting
+ Stackdriver / action audit

Not convinced (low utility and/or high lift):
- API server + cron / task queues
- Database

These already have dedicated tooling and we need pretty granular dynamic control of them.

Reply:
I thought a primary goal of all this was to unify all the deployment tooling under one umbrella. I already have Ruby and shell tooling for all the stackdriver stuff and could easily have extended it for reporting, but I got the impression this was the direction you wanted to go.

I have a prototype for all the stackdriver config for monitoring and action audit and instantiating the MySQL database. I'll push draft reviews for that just go give an idea of what it's looking like.

Reply:
Changes for monitoring are [here (public)](https://github.com/all-of-us/workbench-devops/pull/61/files) and [here (closed)](https://github.com/all-of-us/workbench-devops/pull/61/files).

Reply:
```
I thought a primary goal of all this was to unify all the deployment tooling under one umbrella.
```

I don't think I ever implied this should be a goal. I suggested Terraform as an alternative/uniform solution to two problems which I saw being addressed with lots of custom code: 
1. maintaining Stackdriver metrics / dashboards
2. maintaining sets of BigQuery tables/views

Initializing the entire application environment, or pushing the GAE app using Terraform is not something I considered or meant to encourage. If you see some advantages to doing this which I'm missing, feel free to put together a proposal, but it's definitely not a clear win in my mind. Specifically, outside of consistency I don't see many upsides but I do see a lot of potential complications from doing this, e.g. ensuring we have analogous tooling to support specific flows like rollbacks or traffic switches between versions.

```
Note that so far, all of the modules I'm looking at have some level of tooling.
```

The difference is the amount of custom logic/tooling we've created to maintain those modules, and which can be replaced/simplified, as well as the various dynamic interactions we need to make with that service.